### PR TITLE
zWLM Classification for EJB Timers

### PR DESCRIPTION
**Pull request by mmerali96**
Issue #8855

### CODE REVIEW

**Comment by andrea-pichler** [comment]
No message review requested from L2

File: dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/container/TimerNpRunnable.java [hunk 1]
```diff
@@ -191,29 +191,47 @@ private void doWork() {
         }
     }
 
+    /**
```
Comment (tkburroughs) [inline dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/container/TimerNpRunnable.java]: javadoc is wrong ... "Getter for bean id" is not correct; it is the timer implementation.

Also, if you have an @return there should be text after it to say what is returned.  For simple getters, the method description and @return description are often redundant, so you could have just one or the other.  An empty @return will produce a javadoc warning, so if present, there should be text after it.

File: dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/container/TimerNpRunnable.java [hunk 2]
```diff
@@ -191,29 +191,47 @@ private void doWork() {
...
+     * Getter for bean id
+     *
+     * @return
+     */
+    public TimerNpImpl getNpTimer() {
+        return this.ivTimer;
+    }
+
+    /**
+     * Getter for method id
```
Comment (tkburroughs) [inline dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/container/TimerNpRunnable.java]: if you have an @return there should be text after it to say what is returned. For simple getters, the method description and @return description are often redundant, so you could have just one or the other. An empty @return will produce a javadoc warning, so if present, there should be text after it.

File: dev/com.ibm.ws.ejbcontainer.timer/bnd.bnd [hunk 3]
```diff
@@ -30,8 +30,11 @@ Include-Resource: \
...
 	com.ibm.ws.ejbcontainer;version=latest,\
 	com.ibm.ws.ejbcontainer.core;version=latest,\
-	com.ibm.ws.kernel.service,\
+	com.ibm.ws.kernel.service;version=latest,\
 	com.ibm.ws.logging,\
+	com.ibm.ws.container.service;version=latest,\
 	com.ibm.websphere.org.osgi.core,\
 	com.ibm.websphere.org.osgi.service.component,\
-	com.ibm.wsspi.org.osgi.service.component.annotations
+	com.ibm.wsspi.org.osgi.service.component.annotations,\
```
Comment (tkburroughs) [inline dev/com.ibm.ws.ejbcontainer.timer/bnd.bnd]: indent is wrong here.... not sure if it is a tabs/blanks issue.... but this last line should line up.

Also... last line should NOT end with ,\   I think that could cause problems in the future if anything else were ever added after this that is not part of buildpath.

File: dev/com.ibm.ws.ejbcontainer.timer/resources/OSGI-INF/metatype/metatype.xml [hunk 4]
```diff
@@ -30,6 +30,14 @@
         <AD name="%persistentExecutorRef" description="%persistentExecutorRef.desc"
             id="persistentExecutorRef" type="String" required="false" ibm:type="pid" ibm:reference="com.ibm.ws.concurrent.persistent.executor"/>
         <AD id="persistentExecutor.target" type="String"  ibm:final="true" default="(service.pid=${persistentExecutorRef})" name="internal" description="internal use only"/>
+
```
Comment (tkburroughs) [inline dev/com.ibm.ws.ejbcontainer.timer/resources/OSGI-INF/metatype/metatype.xml]: Would be nice if the indent matched the above entries.... probably some tab vs blank problem.  If you just copy the white space from above to your new lines, things should match up.

File: dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/container/TimerNpRunnable.java (hunk 1 above)
Comment (mmerali96) [reply dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/container/TimerNpRunnable.java]: Got it, I updated the comment and removed @return here and in other places.

File: dev/com.ibm.ws.ejbcontainer.timer/bnd.bnd (hunk 3 above)
Comment (mmerali96) [reply dev/com.ibm.ws.ejbcontainer.timer/bnd.bnd]: Made change

File: dev/com.ibm.ws.ejbcontainer.timer/resources/OSGI-INF/metatype/metatype.xml (hunk 4 above)
Comment (mmerali96) [reply dev/com.ibm.ws.ejbcontainer.timer/resources/OSGI-INF/metatype/metatype.xml]: Made change

**Comment by mmerali96** [comment]
Squashed. Running PB #build

**Comment by mmerali96** [comment]
Build was green. Merging PR

**Pull request merged**